src/targets/gpu/include/migraphx/gpu/compiler.hpp

```diff
@@ -66,16 +68,30 @@ struct compiler_replace
     }
 };
 
-using compiler_compile = std::function<compiler_replace(context&, instruction_ref, operation)>;
+struct tuning_config
+{
+    value problem;
+    std::vector<value> solutions;
+};
+
+using compiler_compile =
```

Comment:
It would be nice to document how the "compile_ops" pass works. It's become complicated.